File: hurdle.py
l=int(input("Enter the array size ::"))
n=int(input("enter the number"))
k=[]
for i in range(2,n+1):
    if(n>1):
        while(n%i==0):
            k.append(i)
            n=n//i
print(k)

q=str(input("Enter the string :"))
k1=[]
q1=int(q[2])
if(q[0]=='!'):
    x=int(q[6])
    r=int(q[4])
    for j in range(0,r):
        if(q1==j):
           k1.append(x)
           q1=q1+1
        else:
            k1.append(" ")
    print(k1)
elif(q[0]=='?'):
    start=int(q[2])
    end=int(q[4])
    c=0
    for i in range(start,end+1):
        for j in range(0,len(k)):
            match=int(k[j])
            if(match==i):
                if(int(k[j])!=int(k[j-1])):
                    c=c+1
    print(c)
                    

File: Hurdle.cpp
#include<stdio.h>
int main()
{
	int nodes,i,j,f,l,w,edges=0,a[200][200];
	printf("Enter the number of nodes: ");
	scanf("%d",&nodes);
	edges = ((nodes*(nodes -1))/2 )* 0.4;
	printf("\nNumber of edge is: %d\n",edges);
	for(i=0;i<=nodes;i++)
	{
		for(j=0;j<=nodes;j++)
		{
			a[i][j]=9999;
		}
	}
	for(i=1;i<=edges;i++)
	{
		printf("\nEnter the first and last vertices of the edge:\n");
		scanf("%d%d",&f,&l);
		printf("\nEnter the weightage:\n");
		scanf("%d",&w);
		a[f][l]=a[l][f]=w;
	}
	printf("\n\nThe adjacency matrix of the graph is:\n\n\n");
	for(i=0;i<nodes;i++)
	{
		printf("\t%d",i);
	}
	for(i=0;i<nodes;i++)
	{
		printf("\n\n\n%d",i);
		for(j=0;j<nodes;j++)
		{
			if(i==j)
			{
				printf("\t0");
			}
			else
			{
				printf("\t%d",a[i][j]);
			}
		}
	}
}
